### PR TITLE
feature:patientlogin

### PR DESCRIPTION
**Pull request by mohamedebrahim2020**

### CODE REVIEW

File: app/Enums/GenderType.php [hunk 1]
```diff
@@ -11,6 +11,6 @@
  */
 final class GenderType extends Enum
 {
-    const MALE =   1;
-    const FEMALE = 2;
+    const MALE =   "1";
+    const FEMALE = "2";
```
Comment (MohannadElemary2) [inline app/Enums/GenderType.php]: Pleas make it integers

File: app/Http/Controllers/PatientController.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    public function __construct(PatientService $patientService)
+    {
+        $this->patientService = $patientService;
+    }
+
+    public function register(PatientRegisterationRequest $request)
+    {
+        $this->patientService->store($request->except('verified_at'));
+        return response()->json([], Response::HTTP_CREATED);
```
Comment (MohannadElemary2) [inline app/Http/Controllers/PatientController.php]: Return Created Resource

File: app/Http/Resources/IndexDoctorResource.php [hunk 3]
```diff
@@ -1,22 +0,0 @@
-<?php
-
-namespace App\Http\Resources;
-
-use Illuminate\Http\Resources\Json\JsonResource;
-
-class IndexDoctorResource extends JsonResource
```
Comment (MohannadElemary2) [inline app/Http/Resources/IndexDoctorResource.php]: Why those files are deleted?

File: app/Repositories/PatientRepository.php [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+   {
+       $patient->verificationCode()->create([
+            'code' => $code,
+            'expired_at' => Carbon::now()->addHour(),
+       ]);
+   }
+
+   public function findPatientByEmail()
+   {
+        $patient = Patient::filter(app(PatientFilters::class))->firstorfail();
```
Comment (MohannadElemary2) [inline app/Repositories/PatientRepository.php]: Use Repository functions

File: app/Services/PatientService.php [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        $this->repository->storeCode($patient, $verificationCode);
+        $patient->notify(new PatientVerificationMail($patient->email, $verificationCode));
+        return $patient;
+    }
+
+    public function checkCode($data)
+    {
+        $patient = $this->repository->findPatientByEmail();
+        if ($patient->verificationCode->code == $data['code'] && Carbon::now()->lessThanOrEqualTo($patient->verificationCode->expired_at)) {
+            $this->repository->update(["verified_at" => Carbon::now()], $patient->id);
```
Comment (MohannadElemary2) [inline app/Services/PatientService.php:38]: Update or delete the code to not be used again

File: app/Traits/LoginTrait.php [hunk 6]
```diff
@@ -7,13 +7,13 @@
...
-        $request->request->add([
+        request()->request->add([
             'grant_type'    => $request->grant_type,
             'client_id'     => $request->client_id,
             'client_secret' => $request->client_secret,
-            'username'      => $request->username,
+            'username'      => $request->email,
             'password'      => $request->password,
             'scope'         => ['*'],
         ]);
```
Comment (MohannadElemary2) [inline app/Traits/LoginTrait.php]: Handle the error returned from passport for any reason

File: database/migrations/2021_01_11_143724_create_patients_table.php [hunk 7]
```diff
@@ -21,7 +21,7 @@ public function up()
             $table->string('phone')->unique();
             $table->string('photo');
             $table->integer('gender');
-            $table->timestamp('verified_at');
+            $table->timestamp('verified_at')->nullable();
```
Comment (MohannadElemary2) [inline database/migrations/2021_01_11_143724_create_patients_table.php:24]: When editing an old migration, make a new migration for the edit

**Comment by MohannadElemary2** [comment]
@mohamedebrahim2020 
- Please pull the develop branch to your branch and reslove conflicts
- Make sure to clone the last develop with the last updates when working on a new feature

File: app/Http/Controllers/PatientController.php (hunk 2 above)
Comment (mohamedebrahim2020) [reply app/Http/Controllers/PatientController.php]: done

File: app/Enums/GenderType.php (hunk 1 above)
Comment (mohamedebrahim2020) [reply app/Enums/GenderType.php]: done

File: app/Http/Resources/IndexDoctorResource.php (hunk 3 above)
Comment (mohamedebrahim2020) [reply app/Http/Resources/IndexDoctorResource.php]: solved by git pull and solving conflicts

File: app/Repositories/PatientRepository.php (hunk 4 above)
Comment (mohamedebrahim2020) [reply app/Repositories/PatientRepository.php]: done

File: database/migrations/2021_01_11_143724_create_patients_table.php (hunk 7 above)
Comment (mohamedebrahim2020) [reply database/migrations/2021_01_11_143724_create_patients_table.php:24]: i will take it in my account after that

File: app/Services/PatientService.php (hunk 5 above)
Comment (mohamedebrahim2020) [reply app/Services/PatientService.php:38]: done

**Pull request merged**